qiskit/qasm/libs/qelib1.inc

```diff
@@ -29,11 +29,11 @@ gate h a { u2(0,pi) a; }
 // Clifford gate: sqrt(Z) phase gate
 gate s a { u1(pi/2) a; }
 // Clifford gate: conjugate of sqrt(Z)
-gate sdg a { u1(-pi/2) a; }
+gate sinv a { u1(-pi/2) a; }
```

Comment:
I dont think the names in the qasm dump should change, since the standard library was no way to deprecated previous names. Maybe create  `qiskit/qasm/libs/qelib2.inc` for that? @ajavadia ?

Reply:
I changed these lines to
```
gate sdg a { u1(-pi/2) a; }
gate sinv a { sdg a; }
```
That should work, wouldn't it?

Reply:
This does not seem to work with `QuantumCircuit.from_qasm_file`. I get 

> Custom non-opaque gates are not supported by ast_to_dag module

Reply:
Oh, yes this happened because the `sdg`/`tdg` were changed to `*inv` in the interpreter. I added both names `dg` and `inv` as standard gate names. Should be fixed by 76a2346.

Reply:
Could you also add a test case which would catch issues like this? Also it may be better if ast_to_dag populated `standard_extension` from qelib1.inc, although that could be saved for a future pr.

Reply:
@ewinston 918c087 implements a test to build a circuit from a qasm file including all gates from qelib1 and compares it against the same circuit set up via circuit methods. That could ideally also be automated by taking the gates from `qelib1.inc` itself. 
I agree ast_to_dag should take the gates from qelib directly, that'd be straightforward for the gate names but probably more difficult to automatically find the corresponding `Gate` for the `standard_extension` dict. Also these gates are not really changing, are they?